File: js/embedding_example.js
const { OpenAI } = require('openai');  

/**
 * Demonstrates the use of the Heurist API to generate embeddings for input text.
 
 * Args:
 * - apiKey (string): The API key required for authentication. 
 * - baseURL (string): The base URL for the Heurist API endpoint.
 * - input (string or array): A single string or an array of strings for which embeddings will be generated.
 *
 * Functionality:
 * - Generates embeddings for the provided input using the specified AI model.
 * - Logs the embedding for each input item if the input is an array.
 * - Displays the number of prompt tokens used during the process.
 
 * API Reference:
 * - model (string): Specifies the model to use for generating embeddings. 
 * - encoding_format (string): Specifies the format of the generated embeddings, e.g., 'float'.
 */


const client = new OpenAI({
  apiKey: "your_user_id#your_api_key",  // For API key, visit: https://dev-api-form.heurist.ai/
  baseURL: 'https://llm-gateway.heurist.xyz',
});
const input = ['hello', 'world'];

async function generateEmbeddings() {
  try {
    const embeddings = await client.embeddings.create({
      model: 'BAAI/bge-large-en-v1.5',          // For supported models, visit: https://github.com/heurist-network/heurist-models/blob/main/models.json
      input: input,
      encoding_format: 'float',
    });

    if (typeof input === 'string') {
      console.log(embeddings.data[0].embedding);
    } else {
      input.forEach((text, index) => {
        console.log(`Embedding for '${text}':`);
        console.log(embeddings.data[index].embedding);
      });
    }

    console.log('Prompt tokens used:', embeddings.usage.prompt_tokens);
  } catch (error) {
    console.error('Error generating embeddings:', error);
  }
}

generateEmbeddings();

File: basic/embeddings/embedding_example.py
from openai import OpenAI
"""
Demonstrates the use of the Heurist API to generate embeddings for input text.

Args:
- api_key (str): The API key required for authentication with the Heurist API.
- base_url (str): The base URL for the Heurist API endpoint.
- input (list or str): A list of strings or a single string for which embeddings will be generated.

Functionality:
- Creates embeddings for the provided input text using a specified model.
- Prints the embedding for each input item if the input is a list.
- Displays the number of prompt tokens used in the request.
"""


api_key = "your_user_id#your_api_key"  # For API key, visit: https://dev-api-form.heurist.ai/
base_url = "https://llm-gateway.heurist.xyz"
client = OpenAI(api_key=api_key, base_url=base_url)

input = ["hello", "world"]

embeddings = client.embeddings.create(
    model="BAAI/bge-large-en-v1.5",  # For the supported model list, visit: https://docs.heurist.ai/dev-guide/supported-models
    input=input,
    encoding_format="float"
)

if isinstance(input, str):
    print(embeddings.data[0].embedding)
else:
    # If input is a list, print the embeddings for each element
    for i in range(len(input)):
        print(f"Embedding for '{input[i]}':")
        print(embeddings.data[i].embedding)

print("Prompt tokens used:", embeddings.usage.prompt_tokens)
